api/base/serializers.py

```diff
@@ -79,22 +79,26 @@ def _tpl(val):
     return None
 
 
+#todo probably needs refactoring
 def _get_attr_from_tpl(attr_tpl, obj):
-    attr_name = _tpl(str(attr_tpl))
-    if attr_name:
-        attribute_value = getattr(obj, attr_name, ser.empty)
-        if attribute_value is not ser.empty:
-            return attribute_value
-        elif attr_name in obj:
-            return obj[attr_name]
-        else:
-            raise AttributeError(
-                '{attr_name!r} is not a valid '
-                'attribute of {obj!r}'.format(
-                    attr_name=attr_name, obj=obj,
-                ))
+    if hasattr(obj, 'pk'):
```

Comment:
@brianjgeiger @barbour-em  Discussed in-person with @PatrickEGorman : This was a misunderstanding in how this function works. L93 was raising an error because you were trying to index into a User object, but we shouldn't even get to that branch (meaning getattr was returning ser.empty). 

To clarify: when using `Link`, the name that goes between `< >` is the attribute that you want to pull off the object, which may be different from the URL parameter.

For example.

```ython
Link('users:user-detail', kwargs={'user_id': '<pk>'})
```